lib/ansible/parsing/splitter.py

```diff
@@ -79,8 +80,7 @@ def parse_kv(args, check_raw=False):
                 k = x[:pos]
                 v = x[pos + 1:]
 
-                # FIXME: make the retrieval of this list of shell/command options a function, so the list is centralized
-                if check_raw and k not in ('creates', 'removes', 'chdir', 'executable', 'warn', 'stdin', 'stdin_add_newline', 'strip_empty_ends'):
+                if check_raw and k not in get_command_args():
```

Comment:
Although the current list of key-value arguments is out-of-date, I don't think that's a good reason to move them into module_utils. The main reason is that the list of valid arguments varies by module. There are currently 6 different modules that make use of `check_raw`:

https://github.com/ansible/ansible/blob/116948cd1468b8a3e34af8e3671f4089e1f7584c/lib/ansible/constants.py#L116-L117

While there are arguments in common between those modules, they're not all the same. For example,
the `win_command` module does not accept `strip_empty_ends`.

It seems like we'd be better off checking the actual list of arguments accepted by the module we're parsing for, and use those when calling `parse_kv`, instead of using `check_raw` to activate a hard-coded list.

Reply:
It looks like the approach in this PR was suggested by @sivel in https://github.com/ansible/ansible/issues/73005#issuecomment-747699681 -- although that seems to have been based on a ~9 year old code comment that predates the split to collections:

https://github.com/ansible/ansible/blob/29aea9ff3466e4cd2ed00524b9e56738d568ce8b/lib/ansible/parsing/splitter.py#L90-L91

Reply:
I understand the concern and I would need some more guidance to resolve this as I am not very familiar with the codebase. I can see that `parse_kv` is used in a few places. Should I try to get the actual list of arguments accepted by the module here?
https://github.com/ansible/ansible/blob/29aea9ff3466e4cd2ed00524b9e56738d568ce8b/lib/ansible/cli/console.py#L204-L206

Reply:
Currently there isn't a good way to get the supported action/module args from the action name while parsing args.
Rather than implementing that (a non-trivial task), one option would be to add `cmd` to the existing hard-coded list used by the splitter when `check_raw` is enabled.

While that would allow resolving https://github.com/ansible/ansible/issues/73005, it would also have the unwanted affect of changing how the `raw` action is handled.

For example, a playbook with a task such as the following would be affected:

```
    - raw: some_command -o cmd=something
```

Currently this results in running: `some_command -o cmd=something`

With `cmd` in the list of args used by `check_raw`, it would instead become: `some_command -o`

This should be discussed further before we commit to a solution.